src/bp/core/services/dialog/decision-engine.ts

```diff
@@ -35,7 +34,39 @@ export class DecisionEngine {
     }
   }
 
-  private async sendSuggestedReply(reply, sessionId, event) {
+  protected _findBestReply(event: IO.IncomingEvent): IO.SuggestedReply | undefined {
+    const replies = _.sortBy(event.suggestedReplies, reply => -reply.confidence)
+    const lastMsg = _.last(event.state.session.lastMessages)
+
+    // If the user asks the same question, chances are he didnt get the response he wanted.
+    // So we cycle through the other suggested replies and return the next best reply with a high enough confidence.
+    for (let i = 0; i < replies.length; i++) {
+      const bestReplyIntent = replies[i].intent
+      const lastMessageIntent = lastMsg && lastMsg.intent
+
+      if (bestReplyIntent === lastMessageIntent) {
+        const nextBestReply = replies[i + 1]
```

Comment:
array potentially out of length

Reply:
in that case `nextBestReply` will be undefined and a condition checks for undefined right after this line.

Reply:
https://stackblitz.com/edit/typescript-aqhf7p